pkg/controllers/consolidation/controller.go

```diff
@@ -468,198 +525,130 @@ func (c *Controller) calculateLifetimeRemaining(node candidateNode) float64 {
-		}
-		return true
-	})
-	// We do one final check to ensure that the node that we are attempting to consolidate isn't
-	// already handled for deletion by some other controller. This could happen if the node was markedForDeletion
-	// between returning the candidateNodes and getting the stateNodes above
-	if candidateNodeIsDeleting {
-		return consolidationAction{result: consolidateResultNoAction}, nil
-	}
-
-	// We get the pods that are on nodes that are deleting
-	deletingNodePods, err := nodeutils.GetNodePods(ctx, c.kubeClient, lo.Map(markedForDeletionNodes, func(n *state.Node, _ int) *v1.Node { return n.Node })...)
+	newNodes, allPodsScheduled, err := c.simulateScheduling(ctx, node)
 	if err != nil {
-		return consolidationAction{result: consolidateResultUnknown}, fmt.Errorf("failed to get pods from deleting nodes, %w", err)
-	}
-	pods := append(node.pods, deletingNodePods...)
-	scheduler, err := c.provisioner.NewScheduler(ctx, pods, stateNodes, scheduling.SchedulerOptions{
-		SimulationMode: true,
-	})
-
-	if err != nil {
-		return consolidationAction{result: consolidateResultUnknown}, fmt.Errorf("creating scheduler, %w", err)
+		// if a candidate node is now deleting, just retry
+		if errors.Is(err, errCandidateNodeDeleting) {
+			return lifecyleCommand{action: lifecycleActionDoNothing}, nil
+		}
+		return lifecyleCommand{}, err
 	}
 
-	newNodes, inflightNodes, err := scheduler.Solve(ctx, pods)
-	if err != nil {
-		return consolidationAction{result: consolidateResultUnknown}, fmt.Errorf("simulating scheduling, %w", err)
+	// if not all of the pods were scheduled, we can't do anything
+	if !allPodsScheduled {
+		return lifecyleCommand{action: lifecycleActionNonePossible}, nil
 	}
 
 	// were we able to schedule all the pods on the inflight nodes?
 	if len(newNodes) == 0 {
-		schedulableCount := 0
-		for _, inflight := range inflightNodes {
-			schedulableCount += len(inflight.Pods)
-		}
-		if len(node.pods) == schedulableCount {
-			return consolidationAction{
-				oldNodes:       []*v1.Node{node.Node},
-				disruptionCost: disruptionCost(ctx, node.pods),
-				result:         consolidateResultDelete,
-			}, nil
-		}
+		return lifecyleCommand{
+			nodesToRemove: []*v1.Node{node.Node},
+			action:        lifecycleActionDelete,
+		}, nil
 	}
 
 	// we're not going to turn a single node into multiple nodes
 	if len(newNodes) != 1 {
-		return consolidationAction{result: consolidateResultNotPossible}, nil
+		return lifecyleCommand{action: lifecycleActionNonePossible}, nil
 	}
 
 	// get the current node price based on the offering
 	// fallback if we can't find the specific zonal pricing data
 	offering, ok := cloudprovider.GetOffering(node.instanceType, node.capacityType, node.zone)
 	if !ok {
-		return consolidationAction{result: consolidateResultUnknown}, fmt.Errorf("getting offering price from candidate node, %w", err)
+		return lifecyleCommand{action: lifeCycleActionUnknown}, fmt.Errorf("getting offering price from candidate node, %w", err)
```

Comment:
This seems like it was implemented like this previously, but shouldn't this be a `lifeCycleActionFailed` and we should reserve the Unknown action for any other unexpected errors? It looks like we'd never return that if not here anyway.